clang/lib/CodeGen/CGHLSLRuntime.h

```diff
@@ -125,15 +125,6 @@ class CGHLSLRuntime {
   // End of reserved area for HLSL intrinsic getters.
   //===----------------------------------------------------------------------===//
 
-  struct BufferResBinding {
```

Comment:
May I ask how the buffer register number `2` is being stored when this goes away?

Reply:
The register number and space are used in the `llvm.dx.resource.handlefrombinding` intrinsic call when the code to initialize the resource is generated. Then in the LLVM backed the `DXILResourceBindingAnalysis` analyzes the code and collects information about all the shader resources and their bindings based on these calls. This info is then used by other LLVM passes to make sure we generate the correct "dx.resources" metadata in the DXIL container.